lib/dialects/abstract/query-generator.js

```diff
@@ -1192,9 +1192,10 @@ var QueryGenerator = {
                          primaryKeysLeft[0]
 
             , tableRight = as
+            , keyRight  = association.options ? association.options.targetKey || primaryKeysRight[0] : primaryKeysRight[0]
             , attrRight = association.associationType !== 'BelongsTo' ?
                           association.identifierField || association.identifier :
-                          right.rawAttributes[primaryKeysRight[0]].field || primaryKeysRight[0]
+                          right.rawAttributes[keyRight].field || primaryKeysRight[0]
```

Comment:
Do we need key right?
targetKey defaults to the primaryKeyAttribute right?

So you can just do:

```js
attrRight = association.associationType !== 'BelongsTo' ?
                  association.identifierField || association.identifier :
                  right.rawAttributes[association.targetKey].field
```

For clearer code